### PR TITLE
Delete source for documentation that has been moved to docs platform

### PR DESCRIPTION
**Pull request by rachel-mack**
I've rewritten the index page for the mongocxx index to direct people to [new C++ driver docs](https://www.mongodb.com/docs/languages/cpp/drivers/current/), and deleted the transferred source code.

### CODE REVIEW

**Comment by rachel-mack** [comment]
Per a conversation with Ezra, it doesn't look like the test failures are related to the files I've deleted.

**Review by kevinAlbs — CHANGES_REQUESTED** [review]
Thank you for the PR. I expect this will result in 404's for previously working links. I added a custom 404 page on the `gh-pages` branch to link to the new documentation. Example: https://mongocxx.org/notfound

---

Here are other requested changes:

Remove other no-longer-needed pages:

- content/mongocxx-v3/api-abi-versioning.md
- content/mongocxx-v3/installation
- content/mongocxx-v3/policies

Update other references of `https://mongocxx.org/` (excluding referencs to Doxygen API docs: `https://mongocxx.org/api`) to refer to the new pages. There are references in README.md, and in code ([example](https://github.com/mongodb/mongo-cxx-driver/blob/2709febdbc4bbfce609722aed399348a1793fb66/examples/mongocxx/tutorial.cpp#L3)).

File: etc/releasing.md [hunk 1]
```diff
@@ -214,22 +214,31 @@ git push --set-upstream origin releases/v1.2
...
+   documentation](https://github.com/mongodb/docs-cpp/blob/master/source/installation/advanced.txt)
+  to reflect libmongoc requirements
+  - the installation pages for:
+
+    - [Linux](https://github.com/mongodb/docs-cpp/blob/master/source/installation/linux.txt)
+    - [Windows](https://github.com/mongodb/docs-cpp/blob/master/source/installation/windows.txt)
+    - [macOS](https://github.com/mongodb/docs-cpp/blob/master/source/installation/macos.txt)
+
    If the release was not a release candidate, update `Step 2` to reflect the
    new latest stable version to download.
```
Comment (kevinAlbs) [inline etc/releasing.md:243]: Suggest limiting this step to stable (not release candidate) releases. Suggest recommending creating DOCSP ticket to simplify the process. Note updating version in snooty.toml and home page table:

```suggestion
- If the release was not a release candidate, ensure a [DOCSP JIRA ticket](https://jira.mongodb.org/projects/DOCSP/issues/) is created to request updating:

  - the `Installing the MongoDB C driver` section of the [Advanced Installation documentation](https://github.com/mongodb/docs-cpp/blob/master/source/installation/advanced.txt) to reflect libmongoc requirements
  - the `Driver Status by Family and Version` section of the [home page](https://github.com/mongodb/docs-cpp/blob/master/source/installation/linux.txt)
  - the [C++ driver version](https://github.com/mongodb/docs-cpp/blob/e13d787967172220ae19e5d78df61e2071735f0f/snooty.toml#L20-L21).
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with a tweak to the release steps.

**Pull request merged**